packages/core/src/extensions/FullySelectedNode/FullySelectedNodeExtension.ts

```diff
@@ -0,0 +1,72 @@
+import { Editor, Extension } from "@tiptap/core";
+import { getBlockInfoFromPos } from "../../api/getBlockInfoFromPos";
+
+// Removes the `ProseMirror-fullyselected` class name from the editor when a
+// NodeSelection is active on a block without inline content, but the DOM
+// selection is within the node, rather than fully wrapping it. These 2
+// scenarios look identical in the editor state, so we need to check the DOM
+// selection to differentiate them.
+const onSelectionChange = (editor: Editor) => {
+  const selection = document.getSelection();
+  if (selection === null) {
+    return;
+  }
+
+  // selectionchange events don't bubble, so we have to scope them in this way
+  // instead of setting the listener on the editor element.
+  if (
+    !editor.view.dom.contains(selection.anchorNode) ||
+    !editor.view.dom.contains(selection.focusNode)
+  ) {
+    return;
+  }
+
+  // Node selection is active.
+  const isNodeSelection = "node" in editor.state.selection;
+  if (!isNodeSelection) {
```

Comment:
this check seems easier than the code above. move to top? (even before document.getSelection?)